### ISSUE TITLE
RCS1234 false positive

### ISSUE THREAD

**Issue by btnlq** [JosefPihrt/Roslynator#593]
**Product and Version Used**: 2.1.3

**Steps to Reproduce**:
```csharp
enum MyEnum
{
    A,
    B = A
}
```

**Actual Behavior**:
Suggests
_RCS1234. Duplicate enum value.
Change enum value to 'B':_
```csharp
enum MyEnum
{
    A = B,
    B = A
}
```
It does not compile.

**Expected Behavior**:
Suggests nothing.
